screens/Directions.js

```diff
@@ -0,0 +1,305 @@
+import React, { useEffect, useRef } from "react";
+import { StyleSheet, TouchableOpacity} from "react-native";
+import MapView, { PROVIDER_GOOGLE, Polyline, Circle } from "react-native-maps";
+import { View, Text, Icon } from "native-base";
+import CurrentLocationButton from "../components/CurrentLocationButton";
+import HTML from "react-native-render-html";
+import PropTypes from "prop-types";
+
+
+/**
+ * TODO: A) If we want to enabled a dark mode. https://github.com/react-native-community/react-native-maps#customizing-the-map-style
+*/
+/**
+ * US1 - As a user, I would like to navigate through SGW campus.
+ * US2 - As a user, I would like to navigate through Loyola campus.
+ * TODO: There are other user stories to refer.
+ * This is our main screen which includes all the components inside a map.
+ */
+function Directions (props) {
+
+    const [instructionIndex, setInstructionIndex] = React.useState(0);
+    const [isLastInstruction, setIsLastInstruction] = React.useState(false);
+    const [isFirstInstruction, setIsFirstInstruction] = React.useState(false);
+    const mapRef = useRef(null);
+ 
+    /* 2. Read the params from the navigation state */
+    const { params } = props.navigation.state;
+    const destinationResponse = params ? params.destinationResponse : null;
+
+    useEffect(() => {
```

Comment:
Isn't this true by default? When you search for directions, trivially, the first instruction that pops up is the first instruction

Reply:
Good Point! I'll try it out by removing this useEffect.

Reply:
Good call!

Reply:
Update, that useEffect had a purpose. The arrow button were always disabled one click later, i.e You are at the last instruction but you can still select next and only _then_ the button will get disabled. Thus, I added a condition for the last instruction in that useEffect. This useEffect has a purpose so I kept it. However, I initialized the isFirstInstruction to true!